scripts/EventManager.js

```diff
@@ -0,0 +1,115 @@
+import {Editor} from './../index'
+import VoiceEvent from "./domain/VoiceEvent";
+
+export default class EventManager {
+
+    constructor() {
+        this.editor = new Editor();
+        this.events = [];
+    }
+
+    init() {
+        this.editor.addEvent('result', speech => {
+            this.recognizeCommand(speech)
+        });
+    }
+
+    startRecognition() {
+        this.editor.start();
+    }
+
+    stopRecognition() {
+        this.editor.stop();
+    }
+
+    addEvent(voiceCommand, callback, args) {
+
+        let voiceEvent = this.events.find(item => item.equals(voiceCommand));
+
+        if (!voiceEvent) {
+            this.events.push(EventManager.buildVoiceEvent(voiceCommand, callback, args));
+        } else {
+            //TODO: check if nested events changed
+            voiceEvent.addAction(callback);
+            this.events[EventManager.findIndex(this.events, voiceCommand)] = voiceEvent;
```

Comment:
Why do you rewrite full object? You need only add action(line 33). Remove this line